tilelang/contrib/msvc.py

```
@@ -0,0 +1,290 @@
+def _patch_source_for_msvc(path: str, tmp_dir: str) -> str:
+    if os.path.splitext(path)[1].lower() not in (".c", ".cc", ".cpp"):
+        return path
+
+    with open(path, encoding="utf-8") as src:
+        source = src.read()
+
+    patched = _ALIGN_ATTRIBUTE_RE.sub(r"__declspec(align(\1))", source)
+    if patched == source:
+        return path
+
+    prefix = hashlib.md5(os.path.dirname(os.path.abspath(path)).encode()).hexdigest()[:8]
```

Ruff (0.15.11)

[error] Probable use of insecure hash functions in `hashlib`: `md5`

(S324)